Update docs

luu-alex opened the pull request:
## Description
Updating documents with relevant changes

Please include a summary of the changes and be sure to follow our [Contribution Guidelines](../CONTRIBUTIONS.md).

<!--
Optional if an issue is fixed:
Fixes #(issue)
-->

## Type of change

<!-- Please delete options that are not relevant. -->


## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] Any dependent changes have been merged and published in downstream modules.
- [x] I have updated the `CHANGELOG.md` file in the root folder.

exx8 committed:
add nonce to send options.

exx8 committed:
add nonce to send options.

exx8 committed:
Update CHANGELOG.md

exx8 committed:
linting

exx8 committed:
Merge remote-tracking branch 'origin/1.x' into 1.x

exx8 committed:
merge

Commit:
updating docs

starwalker00 committed:
update web3-shh.rst : whisper-overview broken link (#4170)

whisper overview has been moved from https://github.com/ethereum/go-ethereum/wiki/Whisper to https://eth.wiki/concepts/whisper/whisper-overview

mongolsteppe committed:
[Docs] Fixed a broken link (#4151)

Original link gives out a 404, replaced by the closest thing I could find.

blackblather committed:
Possible typo in docs (#4088)

I think the author missed a "this" in the phrase

Spacesai1or commented on CHANGELOG.md:
```diff
@@ -400,4 +400,7 @@ Released with 1.0.0-beta.37 code base.
 - Removing the underscore package 
 
+### Changed
+ - Docs are being updated  
```
```suggestion
 - Grammar fix and updated Swarm and Whisper doc links
```

fulldecent committed:
Typically you will have only one Web3 connection, use const (#3967)

Co-authored-by: Gregory Markou <16929357+GregTheGreek@users.noreply.github.com>

evertonfraga committed:
Removing deprecation notice for HttpProvider (#4008)

* Removing deprecation notice for HttpProvider

From my view, it is just a provider with less capabilities than websockets, but still widely useful — and widely used as well.

@frozeman @nivida @GregTheGreek any thoughts?

* Update include_package-core.rst

* Update CHANGELOG.md

Co-authored-by: Gregory Markou <16929357+GregTheGreek@users.noreply.github.com>
Co-authored-by: Alex <alex.luu@mail.utoronto.ca>

mongolsteppe committed:
[Docs] Updated solidity example to modern syntax (#4147)

* [Docs] Updated solidity example to modern syntax

Replaced the old constructor function syntax with the modern one, added 'emit' to event calls, added pragma, added a valid bytes32 value and updated the JSON ABI.

* [Docs] Fixed a broken link (#1)

Original link gives out a 404, replaced by the proper guide.

Co-authored-by: Juan Alonso <Juan.Alonso@trimble.com>

* Revert "[Docs] Fixed a broken link (#1)"

This reverts commit 0de1272d062ea20daf47d3a7e0b39789a74e1130.

Co-authored-by: Juan Alonso <Juan.Alonso@trimble.com>

Commit:
Merge branch '1.x' of https://github.com/exx8/web3.js into alex/update-docs

Spacesai1or committed:
Update CHANGELOG with next anticipated version

Spacesai1or committed:
npm i and build for v1.5.0-rc.0

Spacesai1or committed:
v1.5.0-rc.0

Commit:
added EIP-2718 and EIP-1559 documentation

Commit:
fixing typo

Spacesai1or commented on docs/web3-eth.rst:
```diff
@@ -1419,17 +1421,21 @@ Parameters
     - ``gasPrice`` - ``Number|String|BN|BigNumber``: (optional) The price of gas for this transaction in :ref:`wei <what-is-wei>`, defaults to :ref:`web3.eth.gasPrice <eth-gasprice>`.
+    - ``type`` - ``Number|String|BN|BigNumber``: (optional) A positive unsigned 8-bit number between 0 and 0x7f that represents the type of the transaction. 
+    - ``maxFeePerGas`` - ``Number|String|BN|BigNumber``: (optional) The maximum fee per gas that the transaction is willing to pay in total
+    - ``maxPriorityFeePerGas`` - ``Number|String|BN|BigNumber`` (optional) The maximum fee per gas to give miners to incentivize them to include the transaction (Priority fee)
```
```suggestion
    - ``maxPriorityFeePerGas`` - ``Number|String|BN|BigNumber`` (optional, defaulted to ``1 Gwei``) The maximum fee per gas to give miners to incentivize them to include the transaction (Priority fee)
```

Spacesai1or commented on docs/web3-eth.rst:
```diff
@@ -1419,17 +1421,21 @@ Parameters
     - ``gas``  - ``Number``: (optional, default: To-Be-Determined) The amount of gas to use for the transaction (unused gas is refunded).
     - ``gasPrice`` - ``Number|String|BN|BigNumber``: (optional) The price of gas for this transaction in :ref:`wei <what-is-wei>`, defaults to :ref:`web3.eth.gasPrice <eth-gasprice>`.
+    - ``type`` - ``Number|String|BN|BigNumber``: (optional) A positive unsigned 8-bit number between 0 and 0x7f that represents the type of the transaction. 
+    - ``maxFeePerGas`` - ``Number|String|BN|BigNumber``: (optional) The maximum fee per gas that the transaction is willing to pay in total
```
```suggestion
    - ``maxFeePerGas`` - ``Number|String|BN|BigNumber``: (optional, defaulted to ``(2 * block.baseFeePerGas) + maxPriorityFeePerGas``) The maximum fee per gas that the transaction is willing to pay in total
```

Commit:
addressing feedback

Spacesai1or commented on docs/web3-eth.rst:
```diff
@@ -1419,17 +1421,21 @@ Parameters
     - ``gas``  - ``Number``: (optional, default: To-Be-Determined) The amount of gas to use for the transaction (unused gas is refunded).
     - ``gasPrice`` - ``Number|String|BN|BigNumber``: (optional) The price of gas for this transaction in :ref:`wei <what-is-wei>`, defaults to :ref:`web3.eth.gasPrice <eth-gasprice>`.
+    - ``type`` - ``Number|String|BN|BigNumber``: (optional) A positive unsigned 8-bit number between 0 and 0x7f that represents the type of the transaction. 
+    - ``maxFeePerGas`` - ``Number|String|BN|BigNumber``: (optional, defaulted to ``(2 * block.baseFeePerGas) + maxPriorityFeePerGas``) The maximum fee per gas that the transaction is willing to pay in total
```
```suggestion
    - ``maxFeePerGas`` - ``Number|String|BN``: (optional, defaulted to ``(2 * block.baseFeePerGas) + maxPriorityFeePerGas``) The maximum fee per gas that the transaction is willing to pay in total
```

Spacesai1or replied:
We don't do any conversion to this property if provided by user, which means we only support the type supported by `@ethereumjs/tx` which is [BNLike](https://github.com/ethereumjs/ethereumjs-monorepo/blob/master/packages/util/src/types.ts#L9) as defined [here](https://github.com/ethereumjs/ethereumjs-monorepo/blob/master/packages/tx/src/types.ts#L195)

Spacesai1or commented on docs/web3-eth.rst:
```diff
@@ -1419,17 +1421,21 @@ Parameters
     - ``gasPrice`` - ``Number|String|BN|BigNumber``: (optional) The price of gas for this transaction in :ref:`wei <what-is-wei>`, defaults to :ref:`web3.eth.gasPrice <eth-gasprice>`.
+    - ``type`` - ``Number|String|BN|BigNumber``: (optional) A positive unsigned 8-bit number between 0 and 0x7f that represents the type of the transaction. 
+    - ``maxFeePerGas`` - ``Number|String|BN|BigNumber``: (optional, defaulted to ``(2 * block.baseFeePerGas) + maxPriorityFeePerGas``) The maximum fee per gas that the transaction is willing to pay in total
+    - ``maxPriorityFeePerGas`` - ``Number|String|BN|BigNumber`` (optional, defaulted to ``1 Gwei``) The maximum fee per gas to give miners to incentivize them to include the transaction (Priority fee)
```
```suggestion
    - ``maxPriorityFeePerGas`` - ``Number|String|BN`` (optional, defaulted to ``1 Gwei``) The maximum fee per gas to give miners to incentivize them to include the transaction (Priority fee)
```

Spacesai1or replied:
We don't do any conversion to this property if provided by user, which means we only support the type supported by `@ethereumjs/tx` which is [BNLike](https://github.com/ethereumjs/ethereumjs-monorepo/blob/master/packages/util/src/types.ts#L9) as defined [here](https://github.com/ethereumjs/ethereumjs-monorepo/blob/master/packages/tx/src/types.ts#L191)

Commit:
changing possible types for maxPriorityFeePerGas

Commit:
updating maxFeePerGas type

Commit:
Merge branch 'release/1.5.0' into alex/update-docs

Spacesai1or commented on CHANGELOG.md:
```diff
@@ -400,12 +400,21 @@ Released with 1.0.0-beta.37 code base.
 
 - Removing the underscore package 
 
+=======
```
```suggestion

```

Merge conflict indicator

Commit:
Update CHANGELOG.md

remove merge conflict indicator

Spacesai1or reviewed (changes requested):
Should not be merge in current state

~`alex/update-docs` was branched off of `release/1.5.0` and needs to be merged into `1.x`, so that `1.x` can be merged into `release/1.5.0`. All changes should be replicated in a branch that's based off of `1.x`~

[This](https://github.com/ChainSafe/web3.js/pull/4185/commits/a73c5e92b57bebe2ad9c625d375d4b365ff081fe) commit needs to be reverted
